src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/Orderly.kt

```diff
@@ -10,21 +10,24 @@ class Orderly(val isReviewer: Boolean,
               val isGlobalReader: Boolean,
               val reportReadingScopes: List<String> = listOf(),
               val reportRepository: ReportRepository = OrderlyReportRepository(isReviewer, isGlobalReader, reportReadingScopes),
-              val artefactRepository: ArtefactRepository = OrderlyArtefactRepository(),
               val tagRepository: TagRepository = OrderlyWebTagRepository()) : OrderlyClient
 {
-    constructor(context: ActionContext) : this(context.isReviewer(), context.isGlobalReader(), context.reportReadingScopes)
+    constructor(
+        context: ActionContext
+    ) : this(context.isReviewer(), context.isGlobalReader(), context.reportReadingScopes)
 
     override fun getAllReportVersions(): List<ReportVersionWithDescCustomFieldsLatestParamsTags>
     {
         val basicVersions = reportRepository.getAllReportVersions()
         return mapToReportVersions(basicVersions)
     }
 
-    override fun getDetailsByNameAndVersion(name: String, version: String): ReportVersionWithArtefactsDataDescParamsResources
+    override fun getDetailsByNameAndVersion(
+        name: String,
+        version: String, artefacts: List<Artefact>
+    ): ReportVersionWithArtefactsDataDescParamsResources
```

Comment:
I know this isn't ready for review yet, but I think we've approached this the wrong way - really we should be replacing this whole function with a single call to `orderly.server` to get the report details object, otherwise we'll be making 6 separate calls to `orderly.server`.

Reply:
This is stale, I am definitely trashing this PR as soon as I get a replacement in.

Reply:
> Ok, shall I implement this whole method as an endpoint? Like
> `GET /reports/<name>/versions/<id>/details` ?



> Ok, shall I implement this whole method as an endpoint? Like
> `GET /reports/<name>/versions/<id>/details` ?

Yes, please 👍